### DISCUSSION TITLE
Problem on building app: "Unable to update details on stub app"

### DISCUSSION THREAD

**Discussion by Varisox137** [beeware/briefcase#1404] [Q&A] [answered]
Recently i frequently updated the code of my app and tried to re-build it on multi-platforms, using "briefcase build -u" and "briefcase build android -u". However, the former just couldn't work anymore (it still works n the last version of my code and build config). Meanwhile, i update the android app apk successfully. Kinda weird hah?
The related console output says "Unable to update details on stub app", which i currently have no idea with. Reading the log, seems to have come from subprocess.CalledProcessError with return_code=1.
Much confused...... Any hint on that?

**Comment by Varisox137** [comment]
Now things getting even weirder：
the updating sometimes work and sometimes don't...... but still for Windows building only. Android updates all-fine.
could it be Internet connection reasons?
btw what are "stub apps"......

**Reply by mhsmith** [reply]
The "stub app" on WIndows is a small executable file that does nothing except start Python and run your code.

Please attach the error log file to this issue by dragging it into the box below.

**Reply by Varisox137** [reply]
here's the log...
[briefcase.2023_08_15-04_04_57.build.log](https://github.com/beeware/briefcase/files/12339576/briefcase.2023_08_15-04_04_57.build.log)

**Reply by Varisox137** [reply]
still btw, only noticed now that the log says "OS Release：Win10" but actually i'm using win11 ...? seems buggy

**Reply by rmartin16** [reply]
> still btw, only noticed now that the log says "OS Release：Win10" but actually i'm using win11 ...? seems buggy

This was resolved for Python 3.12 in python/cpython#96289. Unclear if it will be backported to earlier versions of Python. Nonetheless, for most cases, builds after 22000 should be interpreted as Windows 11, I believe...

**Comment by rmartin16** [comment]
Heuristic-based analysis by antivirus software has been known to be triggered by RCEdit.

If you're running antivirus, can you try again with it disabled or with an exclusion for the file path below?

`C:/Users/Varis/AppData/Local/BeeWare/briefcase/Cache/tools/rcedit-x64.exe`

**Reply by Varisox137** [reply]
until now, i haven't reproduced the error again (no matter with antivirus-thing on/off )
sigh......
might just keep using until problems emerge another day

**Comment by pb-413** [comment]
Hello, I've run into this as well during [Tutorial 7](https://docs.beeware.org/en/latest/tutorial/tutorial-7.html), [Updating dependencies](https://docs.beeware.org/en/latest/tutorial/tutorial-7.html).

`briefcase update` works OK, then failed on `briefcase build`; then after finding this page to document the issue, I also had it succeed, so the reproducibility of this seems to be not 100%.

Here was my last log:
[briefcase.2024_05_22-15_51_34.run.log](https://github.com/beeware/briefcase/files/15408167/briefcase.2024_05_22-15_51_34.run.log)
```
> briefcase run -u -r
...
Setting stub app details... started
Fatal error: Unable to commit changes
```

Strange. Taking note and will return if it happens again.

**Reply by pb-413** [reply]
Ran into a similar issue during [Tutorial 9](https://docs.beeware.org/en/latest/tutorial/tutorial-9.html), [Runtime tests](https://docs.beeware.org/en/latest/tutorial/tutorial-9.html#runtime-tests).

Log snippet:
```
> briefcase run --test -r
[helloworld] Building App...
Setting stub app details... started
Fatal error: Unable to commit changes
Setting stub app details... errored
```

Full log: 
[briefcase.2024_05_23-09_15_37.run.log](https://github.com/beeware/briefcase/files/15417466/briefcase.2024_05_23-09_15_37.run.log)

The only similarity I can see so far is that I am trying to complete the action with a one-liner in the past two logs, but I thought I also tried with more atomic actions in my previous comment. Able to reproduce that just now: 
[briefcase.2024_05_23-09_31_07.build.log](https://github.com/beeware/briefcase/files/15417541/briefcase.2024_05_23-09_31_07.build.log)

Able to do that about 5 times in a row in my current state, so it doesn't look like it is rare but possibly fickle?

Nothing relevant in event viewer.

**Reply by freakboy3742** [reply, marked as answer]
To clarify: anyone having this problem - do you have a third party virus or malware checker installed? 

This is an important detail; unfortunately, what Briefcase is doing at this step can be misread as malicious activity. It's *not* doing anything malicious, and we're using a tool that is widely used for exactly what we're using it for - but virus checkers routinely throw a false positive. There is very little we can do here - the issue is the virus checker.

The only real alternatives are (a) disable the virus checker, or (b) use the visual studio backend. This is harder to set up, and slower to generate results, as you need a visual studio install and do a full project build; but it should avoid the "malicious" activity.

**Reply by pb-413** [reply]
Thank you, this was my next step of investigation. I do have McAfee installed and am looking for their logs.

I might A-B test with and without specific security features enabled.

Will update here with my results / research.

**Comment by puspeshd** [comment]
For me, even after disabling antivirus, it was not working. But it is solved as I just had to add exception for the exe file of my app in QUARANTINE
